libdrgn/python/thread.c

```diff
@@ -0,0 +1,104 @@
+#include "drgnpy.h"
+
+static void Thread_dealloc(Thread *self)
+{
+	drgn_thread_deinit(&self->thread);
+	Py_XDECREF(self->prog);
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static PyObject *Thread_get_tid(Thread *self)
+{
+	return PyLong_FromUnsignedLong(self->thread.tid);
+}
+
+static PyObject *Thread_get_object(Thread *self)
+{
+	if (!(self->prog->prog.flags & DRGN_PROGRAM_IS_LINUX_KERNEL)) {
+		PyErr_SetString(
+			PyExc_AttributeError,
+			"a Thread's associated object is only defined when debugging the kernel");
+		return NULL;
+	}
```

Comment:
This made me realize: we need a `struct drgn_error *drgn_thread_object(struct drgn_thread *thread, const struct drgn_object **ret)` so that libdrgn users can get the object, too. (I think it can just return a pointer to the internal object directly; if the user wants to keep it, they can `drgn_object_copy()` it into their own object.) This check should be in that function.

Reply:
Sure. I added `DRGN_ERROR_INVALID_ATTRIBUTE` and modified `set_drgn_error` accordingly so we could continue to produce the correct Python exception.